backend/ts/src/main/scala/aqua/backend/ts/TypescriptFunc.scala

```diff
@@ -42,6 +42,9 @@ case class TypescriptFunc(func: FuncResolved) {
       .map(_._2)
       .fold("void")(typeToTs)
 
+    val returnVal =
+      if (retType == "void") "Promise.race([promise, Promise.resolve()])" else "promise"
```

Comment:
Take a look at where `retType` is defined and use the `Option` of the return type instead.